invisible_cities/reco/deconv_functions.py

```diff
@@ -0,0 +1,288 @@
+import numpy  as np
+import pandas as pd
+
+from typing  import List
+from typing  import Tuple
+from typing  import Optional
+from typing  import Callable
+
+from scipy        import interpolate
+from scipy.signal import fftconvolve, convolve
+
+from ..core .core_functions import shift_to_bin_centers
+from ..core .core_functions import in_range
+
+def cut_and_redistribute_df(cut_condition : str,
+                            variables     : List[str]=[]) -> Callable:
+    '''
+    Apply a cut condition to a dataframe and redistribute the cut out values
+    of a given variable.
+
+    Parameters
+    ----------
+    df      : dataframe to be cut
+
+    Initialization parameters:
+        cut_condition : String with the cut condition (example "Q > 10")
+        variables     : List with variables to be redistributed.
+
+    Returns
+    ----------
+    pass_df : dataframe after applying the cut and redistribution.
+    '''
+    def cut_and_redistribute(df : pd.DataFrame) -> pd.DataFrame:
+        pass_df = df.query(cut_condition)
+        pass_df._is_copy = False
+        with np.errstate(divide='ignore'):
+            for redist_variable in variables:
+                pass_df[redist_variable] = pass_df[redist_variable] * np.divide(df[redist_variable].sum(), pass_df[redist_variable].sum())
+        return pass_df
+
+    return cut_and_redistribute
+
+def drop_isolated_sensors(distance  : List[float]=[10., 10.],
```

Comment:
This function could be of general use. I will propose to change the name. Something like *select_hits_and_redestribute_signal* and I guess in the variables will be something like "E" or "Ec", right? Maybe given a default could be useful.

Reply:
As I understand this function is doing the same as reco/hits_functions but using directly dataframes instead of HitCollection type. Maybe we can then move this function to reco/hits_functions and eventually rewrite the old code (ie esmeralda) to use this one since it is faster than converting to our HitCollection data type?

Reply:
We have decided to leave that for another PR